bflibrary/src/general/rnc_1fm.c

```diff
@@ -437,14 +437,15 @@ long rnc_unpack(void *packed, void *unpacked, unsigned int flags
     return ret_len;
 }
 
-long UnpackM1(unsigned char *buffer, ulong bufsize)
+i32 UnpackM1(unsigned char *buffer, u32 bufsize)
 {
+    LOGWARN("UnpackM1: (expected size %zu)", bufsize);
```

Comment:
Adding debug output seems to circumvent the segfault race condition/threading/timing caused.